src/schema/v2/NotificationContent/index.ts

```diff
@@ -0,0 +1,178 @@
+          return body ?? []
+        },
+      },
+
+      artworksConnection: {
+        type: artworkConnection.connectionType,
+        args: pageable(),
+        resolve: async ({ object_ids }, args, { artworksLoader }) => {
+          const { page, size } = convertConnectionArgsToGravityArgs(args)
+          const body = await artworksLoader({ ids: object_ids })
+          const totalCount = body.length
+
+          return {
+            totalCount,
+            pageCursors: createPageCursors({ page, size }, totalCount),
+            ...connectionFromArray(body, args),
+          }
+        },
+      },
+    },
+  }
+)
+
+const AlertContentType = new GraphQLObjectType<any, ResolverContext>({
+  name: "AlertNotificationContent",
+  fields: {
+    alert: {
+      type: AlertType,
+      resolve: ({ actor_ids }, _args, { meSearchCriteriaLoader }) => {
+        if (!meSearchCriteriaLoader) {
+          throw new Error("You need to be signed in to perform this action")
+        }
+
+        if (!actor_ids) return null
+
+        const searchCriteriaID = actor_ids[0]
+
+        return meSearchCriteriaLoader(searchCriteriaID)
+      },
+    },
+
+    artworksConnection: {
+      type: artworkConnection.connectionType,
+      args: pageable(),
+      resolve: async ({ object_ids }, args, { artworksLoader }) => {
+        const { page, size } = convertConnectionArgsToGravityArgs(args)
+        const body = await artworksLoader({ ids: object_ids })
+        const totalCount = body.length
+
+        return {
+          totalCount,
+          pageCursors: createPageCursors({ page, size }, totalCount),
+          ...connectionFromArray(body, args),
+        }
+      },
+    },
+  },
+})
+
+const ShowOpenedContentType = new GraphQLObjectType<any, ResolverContext>({
+  name: "ShowOpenedNotificationContent",
+  fields: {
+    show: {
+      type: ShowType,
+      resolve: ({ object_ids }, _args, { showLoader }) => {
+        if (!object_ids) return null
+
+        const showID = object_ids[0]
+
+        return showLoader(showID)
+      },
+    },
+  },
+})
+
+const ViewingRoomPublishedContentType = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "ViewingRoomPublishedNotificationContent",
+  fields: {
+    viewingRoomID: {
+      type: GraphQLNonNull(GraphQLString),
+      description: "The ID of the viewing room, for use in stitching",
+      resolve: ({ object_ids }) => {
+        if (!object_ids) return null
+
+        return object_ids[0]
+      },
+    },
+  },
+})
+
+export const NotificationContentType = new GraphQLUnionType({
+  name: "NotificationContent",
+  types: [
+    ArtworkPublishedContentType,
+    AlertContentType,
+    ArticleFeaturedArtistContentType,
+    ShowOpenedContentType,
+    ViewingRoomPublishedContentType,
+  ],
+  resolveType: ({ activity_type }) => {
+    switch (activity_type) {
+      case "SavedSearchHitActivity":
+        return AlertContentType
```

Comment:
Why not match types 1:1 with their associated `activity_type`? That could solve the naming confusion. 

```
case "SavedSearchHitActivity": 
  return SavedSearchHitActivityType
```

Reply:
At least for `SavedSearch` naming, that's legacy naming and we prefer to always use `Alert` in the schema/front-end now, generally.